src/emm/services/modules_service.py

```diff
@@ -225,9 +230,99 @@ def git_commit_modules(self, commit: str) -> None:
+                all_pull_requests.append(pr_result)
+            else:
+                pr_links = "\n".join(
+                    "{} pr: {}".format(repo, link)
+                    for repo, link in comments.items()
+                    if repo != "base"
+                )
+                self.github_service.pr_comment(pr_url, pr_links)
+                pr_result = f"Base pull request: {pr_url}"
+                all_pull_requests.append(pr_result)
+        return all_pull_requests
+
+    def check_push_branch_to_remote(self, directory: Optional[Path] = None) -> str:
+        """
+        Determine if need push current branch to remote.
+
+        :param directory: Directory to execute the command at.
+        :return: Errors that occur during push branch to remote.
+        """
+        branch = self.git_service.current_branch(directory)
+        if not self.git_service.current_branch_exist_on_remote(branch, directory):
+            return self.git_service.push_branch_to_remote(directory)
+        return ""
+
+    def module_pull_request(self, args: List[str], comment_dict: Dict[str, str]) -> None:
+        """
+        Create pull request for each module.
+
+        :param args: Arguments to pass to the github CLi.
+        :param comment_dict: Dictionary of module and its pull request url.
+        """
+        enabled_modules = self.get_all_modules(True)
+        for module in enabled_modules:
+            module_data = self.get_module_data(module)
+            module_location = Path(module_data.prefix) / module
+            basename = self.git_service.get_base_name(module_location)
+            if basename and self.git_service.check_changes(basename, module_location):
+                error = self.check_push_branch_to_remote(module_location)
+                if error:
+                    raise EnvironmentError(f"{module} could not push to remote.")
+
+                link = self.github_service.pull_request(args, module_location)
+                comment_dict[module] = link
+
+    def base_pull_request(self, args: List[str], comment_dict: Dict[str, str]) -> None:
+        """
+        Create pull request for base repo.
+
+        :param args: Arguments to pass to the github CLi.
+        :param comment_dict: Dictionary of base repo and its pull request url.
+        """
+        err = self.check_push_branch_to_remote()
+        if err:
+            raise EnvironmentError("Base branch could not push to remote.")
+
+        pr_link = self.github_service.pull_request(args)
+        comment_dict[BASE_REPO] = pr_link
+
+    def git_pull_request(self, args: List[str]) -> List[str]:
+        """
+        Create pull request for each module.
+
+        :param args: Arguments to pass to the github CLi.
+        :return: List of pull requests being created associate with its link.
+        """
+        comment_dict: Dict[str, str] = {}
+        self.base_pull_request(args, comment_dict)
```

Comment:
I tend to avoid functions that make changes to the arguments passed into them whenever possible. They make it much more difficult to find where data is coming from. It seems like these function could return data for the `comment_dict` and we could update it in this function.

```python
comment_dict = self.base_pull_request(args)
comment_dict = {**comment_dict, **self.module_pull_requests(args)}
```